### PR TITLE
Talk about host specific/independent instead of mutability

### PR DESCRIPTION
**Pull request by duglin**
Cover my action item from this week's OCI call.

Also moved State json out of config.go since its not really part of
the config.json file/config.

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

File: bundle.md [hunk 1]
```diff
@@ -10,17 +10,20 @@ The definition of a bundle is only concerned with how a container, and its confi
...
+When the bundle is packaged up for distribution, this file MUST be included.
 See [`config.json`](config.md) for more details.
 
-2. `runtime.json` : mutable, host dependent configuration.
-This REQUIRED file, which MUST be named `runtime.json`, contains settings that are host specific such as memory limits, local device access and mount sources.
-The goal is that the bundle can be moved as a unit to another runtime and run the same application if `runtime.json` is reconfigured.
+2. `runtime.json` : contains host-specific configuration data.
+This REQUIRED file, which MUST be named `runtime.json`, contains settings that are host specific such as mount sources and hooks.
+The goal is that the bundle can be moved as a unit to another runtime and run the same application once a host-specific `runtime.json` is defined.
+When the bundle is packaged up for distribution, this file MUST NOT be included.
```
Comment (wking) [inline bundle.md]: There was [concern about this from the Solaris folks][1].  I think allowing, but not requiring, `runtime.json` in distributed bundles makes sense.

[1]: https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/TqcUboTh9KI

File: bundle.md [hunk 2]
```diff
@@ -10,17 +10,20 @@ The definition of a bundle is only concerned with how a container, and its confi
...
+1. `config.json` : contains host independent configuration data.
 This REQUIRED file, which MUST be named `config.json`, contains settings that are host independent and application specific such as security permissions, environment variables and arguments.
+When the bundle is packaged up for distribution, this file MUST be included.
 See [`config.json`](config.md) for more details.
 
-2. `runtime.json` : mutable, host dependent configuration.
-This REQUIRED file, which MUST be named `runtime.json`, contains settings that are host specific such as memory limits, local device access and mount sources.
-The goal is that the bundle can be moved as a unit to another runtime and run the same application if `runtime.json` is reconfigured.
+2. `runtime.json` : contains host-specific configuration data.
+This REQUIRED file, which MUST be named `runtime.json`, contains settings that are host specific such as mount sources and hooks.
```
Comment (wking) [inline bundle.md]: Where did memory limits and local device access go?

File: state.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package specs
+
+// State holds information about the runtime state of the container.
+// This information will be stored in a file called `state.json`.
+// The location of this file will be operating system specific. On Linux
+// it will be in `/run/opencontainers/runc/<containerID>/state.json`
+type State struct {
+	// Version is the version of the specification that is supported.
```
Comment (wking) [inline state.go]: I've [wanted `state.go` in its own file][1] for a while now, but:

* It seems like an independent issue from the thrust of this PR, and
* Previous pushback revolved around [limiting the number of `*.go` files][2]

[1]: https://github.com/opencontainers/specs/pull/87#issuecomment-137200444
[2]: https://github.com/opencontainers/specs/pull/87#issuecomment-128137602

**Comment by crosbymichael** [comment]
LGTM

**Comment by vbatts** [comment]
LGTM

**Pull request merged**